compiler/src/dmd/cond.d

```diff
@@ -659,9 +659,9 @@ extern (C++) final class VersionCondition : DVCondition
             case "AVR":
             case "BigEndian":
             case "BSD":
-            case "CppRuntime_libcpp":
+            case "CppRuntime_libcxx":
             case "CppRuntime_DigitalMars":
-            case "CppRuntime_libstdcpp":
+            case "CppRuntime_libstdcxx":
             case "CppRuntime_Microsoft":
             case "CppRuntime_Sun":
```

Comment:
So just to underline something here, we have CppRuntime_...
- DigitalMars
- Microsoft
- Sun
- libcxx
- libstdcxx

Whilst I can agree that calling them Clang and Gcc is a bit vague (although g++ doesn't work with libc++ AFAIK), I'm not initially convinced they should follow a different style either.

The only alternative I can offer is taking inspiration from their respective documentation pages.

- [libstdc++](https://gcc.gnu.org/onlinedocs/libstdc++/) titles itself the **GNU** C++ Library
- [libc++](https://libcxx.llvm.org/) while it does make a reference to other C++ libraries [by vendor](https://libcxx.llvm.org/#history) - **Apache**, **STLport**, ... - it only titles itself as the bland C++ Standard Library. However various package distributors [seem to](https://packages.debian.org/sid/libc++-18-dev) almost universally [call it](https://pkgsrc.se/wip/libc++) the **LLVM** C++ Standard Library.

So I'd err on the side of `CppRuntime_LLVM` and `CppRuntime_GNU`. But happy to be convinced otherwise if there's a compelling reason why explicit library names are the best handles to use.

Thanks!

Reply:
Your proposal is reasonable, I think it's unlike that GNU or LLVM will implement another standard library. Personally, I would not recognize CppRuntime_GNU/LLVM as libstdc++/libc++ as quickly, but it is rare anyway to have to work with this stuff so doesn't really matter.
For `-target=` recognition, I propose we really do recognize libstdc++/libc++ (and not add gnu/llvm because those are very ambiguous in a triple)

Reply:
(in the end, I think we _do_ rely on the exact library, not the vendor, that'd be a reason to name the thing by its name)

Reply:
With the new evidence @ibuclaw provided, I would just err on the side of reverting the initial PR altogether.
I couldn't care less about naming, which is why I was personally okay with your proposal, but will going from `CppRuntime_Gcc` to `CppRuntime_GNU`, etc... really make that big of a difference that the change has value ? I don't really think so.

Reply:
>  I couldn't care less about naming

Really?? Naming is a big part of programming (a key reason why DMD source is hard to understand and actively avoided by me).

> but will going from `CppRuntime_Gcc` to `CppRuntime_GNU`, etc... really make that big of a difference that the change has value ? 

Yes. The current naming is factually incorrect and hurts legibility of druntime code.

If you don't care about naming, then you also don't have an argument _against_ changing the names, I guess?

Reply:
> Personally, I would not recognize CppRuntime_GNU/LLVM as libstdc++/libc++ as quickly
 
Though it's not our problem, it's libc++ for the Standard C++ library, then there's libstdc++ for the GNU C++ library - a misleading std. :-)

Whichever way we go, my own preference is to keep the version identifiers consistent if at all possible, so either we have:
- CppRuntime_DigitalMars
- CppRuntime_Microsoft
- CppRuntime_Sun
- CppRuntime_LLVM (or Llvm/llvm)
- CppRuntime_GNU (or Gnu/gnu)
or
- CppRuntime_
- CppRuntime_
- CppRuntime_
- CppRuntime_

Reply:
Tradeoff table:
```
        | ..._DigitalMars | ..._DigitalMars | ..._snn       | ..._DigitalMars_snn
        | ..._Microsoft   | ..._Microsoft   | ..._msvc      | ..._Microsoft_msvc
        | ..._Sun         | ..._Sun         | ..._libCstd   | ..._Sun_libCstd
        | ..._Clang       | ..._LLVM        | ..._libcxx    | ..._LLVM_libcxx
        | ..._Gcc         | ..._GNU         | ..._libstdcxx | ..._GNU_libstdcxx
--------|-----------------|-----------------|---------------|---------------------
Clarity |  4              |  7              |  6            |     10
# change|  0              |  2              |  5 (all)      |    5 (all)
backw   |  n/a            | Keep 2 old ones | Keep 5 old?   | Keep 5 old?
compat? |                 |                 |               |
Remark  | incorrect       | assume 1 lib    |               | too verbose?
        |                 | per vendor      |               |
```

minus 1 for clarity of the lib names, because of less known Sun and DM lib names.

Reply:
> Yes. The current naming is factually incorrect and hurts legibility of druntime code.

But the new names hurts readability for (some) people that develop projects outside of DRuntime, which I would hope is most people! Because when you are linking against C++ source code, you know what compiler you use, and so Clang / Gcc are helpful in that regard. For people only linking, I understand the situation may be different.

> If you don't care about naming, then you also don't have an argument against changing the names, I guess?

In this instance, since neither name is completely wrong (IMO, I know you disagree on that point), I care about the amount of change. And I don't think the change from one to the other is materially better. Other people's mileage may vary.

> or it is...
>
> - CppRuntime_snn (guessing here)
> - CppRuntime_msvc
> - CppRuntime_libCstd
> - CppRuntime_libcxx
> - CppRuntime_libstdcxx

That, on the other hand, would be pretty bad.

Reply:
> > Yes. The current naming is factually incorrect and hurts legibility of druntime code.
> 
> But the new names hurts readability for (some) people that develop projects outside of DRuntime, which I would hope is most people! Because when you are linking against C++ source code, you know what compiler you use, and so Clang / Gcc are helpful in that regard. 

This is exactly what is wrong about it!
Let's say you are using Clang, what CppRuntime_* do you most likely need do you think?  Hint: it is CppRuntime_Gcc.

Reply:
> Personally, I would not recognize CppRuntime_GNU/LLVM as libstdc++/libc++ as quickly, but it is rare anyway to have to work with this stuff so doesn't really matter.

WRT one library per vendor, more likely there would be different versions - for example GNU's library is specifically `libstdc++v3`; v2 was pre-GCC 3.x.

> For -target= recognition, I propose we really do recognize libstdc++/libc++ (and not add gnu/llvm because those are very ambiguous in a triple)

Agreed.